WSL/disk-space.md

```diff
@@ -0,0 +1,217 @@
+```
+
+When a distribution is started as read-only, any attempts to write to the filesystem will fail with an error like this:
+
+```bash
+$ touch file
+touch: cannot touch 'file': Read-only file system
+```
+
+To repair a disk mount error in WSL, and restore it back to a usable / writeable state again, you can use the `wsl --mount` command to re-mount the disk with the following steps:
+
+1. Shutdown all WSL distributions by opening PowerShell and entering the command:
+
+    ```powershell
+    wsl.exe --shutdown
+    ```
+
+2. Open PowerShell as administrator (in an elevated command prompt) and enter the mount command, replacing `<path-to-ext4.vhdx>` with the path to the distribution's .vhdx file. For help locating this file, see [How to locate the VHD file and disk path for your Linux distribution](#how-to-locate-the-vhdx-file-and-disk-path-for-your-linux-distribution).
+
+    ```powershell
+    wsl.exe --mount <path-to-ext4.vhdx> --vhd --bare
+    ```
+
+3. Use the `wsl.exe lsblk` command from PowerShell to identify the block device name for the distribution (sd[a-z]) and then enter the following command to repair the disk (replacing `<device>` with the correct block device name, like "sdc"). The `e2fsck` command checks ext4 file systems (the type used by distributions installed with WSL) for errors and repairs them accordingly.
+
+    ```powershell
+    wsl.exe sudo e2fsck -f /dev/<device>
+    ```
+
+4. Once the repair is complete, unmount the disk in PowerShell by entering:
+
+    ```powershell
+    wsl.exe --unmount
+    ```
+
+> [!WARNING]
+> You can use the command: `sudo mount -o remount,rw /` to return a read-only distribution to a usable/writable state, but all changes will be in-memory and so will be lost when the distribution is restarted. We recommend using the steps listed above to mount and repair the disk instead.
+
+## How to locate the .vhdx file and disk path for your Linux distribution
+
+If you are having trouble locating the *.vhdx* file for a Linux distribution, go to the directory path `%LocalAppData%\Packages\`. Each Linux distribution that you have installed includes a subfolder specific to that distribution.
+
+For the default Ubuntu distribution, this directory will look something like `CanonicalGroupLimited.UbuntuonWindows_79rhkp1fndgsc`. Inside that directory will be a `LocalState` folder with the `ext4.vhdx` file that stores your data.
+
+The VHD used by WSL 2 utilizes a ext4 format for the file system, not NTFS, so Windows will not natively read this format outside of the WSL environment. (Tools like Disk Manager or diskpart installed on Windows NTFS file system will not work.)
+
+If you are having trouble locating the disk path where the *.vhdx* file is stored for a particular Linux distribution, you can use PowerShell to help you find it. Replace `<distribution-name>` with the actual distribution name and run the following command:
```

Comment:
Let's use the powershell command as the default way of locating disks. Asking users to find applications that "look like" their distribution is a bit lose and might lead to incorrect disks if multiple versions of the same distributions are installed. The below powershell command is guaranteed to be giving the correct result

Reply:
Updated.